wurst/_items/Spears.wurst

```diff
@@ -0,0 +1,99 @@
+package Spears
+import ClosureEvents
+import ITTDummyCaster
+import Orders
+import ClosureTimers
+import PublicLibrary
+
+let ABILITY_SPEAR_HIT_DETECTION = 'A0EH'
+
+enum SpearType
+    DarkSpear
+    UltraPoisonSpear
+    RefinedPoisonSpear
+    PoisonSpear
+    SteelSpear
+    IronSpear
+    Spear
+
+let spellToSpearType = new HashMap<int, SpearType>
+let spearListeners = new HashMap<ITTDummyCaster, EventListener>()
+
+function onSpear(unit caster, unit target, SpearType spearType)
+    let dummy = new ITTDummyCaster
+        ..delay(8)
+        ..owner(caster.getOwner())
+        ..origin(caster.getPos())
+        ..castTarget(ABILITY_SPEAR_HIT_DETECTION, 1, Orders.creepthunderbolt, target)
+
+    bindSpearDamageListener(dummy, target, spearType)
+
+function bindSpearDamageListener(ITTDummyCaster dummyCaster, unit target, SpearType spearType)
+    let listener = EventListener.add(EVENT_UNIT_DAMAGED) ->
+        let damaged = GetTriggerUnit()
+        if damaged == target and GetEventDamageSource() == dummyCaster.getDummy()
+            if spearListeners.has(dummyCaster)
+                destroy spearListeners.getAndRemove(dummyCaster)
+                onSpearImpact(target, spearType)
+
+    spearListeners.put(dummyCaster, listener)
+
+function onSpearImpact(unit target, SpearType spearType)
+    switch spearType
+        case UltraPoisonSpear
+            spawnItemChance(0.5, ITEM_ULTRA_POISON_SPEAR, target.getPos())
+        case RefinedPoisonSpear
+            spawnItemChance(0.5, ITEM_REFINED_POISON_SPEAR, target.getPos())
+        case PoisonSpear
+            spawnItemChance(0.5, ITEM_POISON_SPEAR, target.getPos())
+        case DarkSpear
+            spawnItemChance(0.4, ITEM_DARK_SPEAR, target.getPos())
+            let manaBurnPerTick = 10.
+            ManaBurn(target, manaBurnPerTick)
+            doPeriodicallyCounted(0.75, 4) manaBurnLoop ->
+                ManaBurn(target, manaBurnPerTick)
```

Comment:
Hmm just going to flag this with request changes until council decides on this

Reply:
I think the stdlib2 should have an immediate option for `doPeriodicallyCounted`, so you aren't force to manually call the callback. Thoughts?